### PR TITLE
[XRI3] Updating RiggedHandMeshVisualizer so to not rely on deprecated XRI controller components

### PR DESCRIPTION
**Pull request by AMollis**
# Overview

Updating `RiggedHandMeshVisualizer` so to not rely on deprecated XRI controller components.

This change will require the rig prefab to specify the input reader being used to obtain the selection value for the pinch gesture. Once the controller components are removed from the new rig, I'll update the new rig to hook-up the new `RiggedHandMeshVisualizer`

### CODE REVIEW

File: org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs [hunk 1]
```diff
@@ -60,6 +62,19 @@ public bool ShowHandsOnTransparentDisplays
                  "Generally, maps to something like a glow or an outline color!")]
         private string pinchAmountMaterialProperty = "_PinchAmount";
 
+        [SerializeField]
+        [Tooltip("The input reader user when pinch selecting an interactable.")]
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs:66]:
```suggestion
        [Tooltip("The input reader used when pinch selecting an interactable.")]
```

File: org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs [hunk 2]
```diff
@@ -306,18 +372,54 @@ private bool ShouldRenderHand()
...
+            else
+            {
+#pragma warning disable CS0618 // XRController is obsolete
+                if (controller == null)
+                {
+                    controller = GetComponentInParent<XRController>();
+                }
 
-            if (controller == null || handRenderer == null) { return; }
+                if (controller != null)
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs:414]:
```suggestion
                else
```

File: org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs [hunk 3]
```diff
@@ -306,18 +372,54 @@ private bool ShouldRenderHand()
...
-                controller = GetComponentInParent<XRBaseController>();
+                success = selectInput.TryReadValue(out value);
             }
+            else
+            {
+#pragma warning disable CS0618 // XRController is obsolete
+                if (controller == null)
+                {
+                    controller = GetComponentInParent<XRController>();
+                }
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs:413]:
```suggestion

```

**Review by ms-RistoRK — APPROVED** [review]
Approved with suggestions.

File: org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs (hunk 2 above)
Comment (AMollis) [reply org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs:413]: nope...`controller = GetComponentInParent<XRController>();` can return null still

**Review by shaynie — APPROVED** [review]

File: org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs (hunk 2 above)
Comment (ms-RistoRK) [reply org.mixedrealitytoolkit.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs:413]: Oh right!

**Pull request merged**